bin/pkg.ml

```diff
@@ -156,6 +156,13 @@ module Lock = struct
       Arg.(
         value & flag
         & info [ "all-contexts" ] ~doc:"Generate the lockdir for all contexts")
+    and+ prefer_oldest =
+      Arg.(
+        value & flag
+        & info [ "prefer-oldest" ]
```

Comment:
I think we'd rather have a flag like `--prefer-versions={oldest,newest}` so that we have a way to run the solver regardless what the default might be.